kubernetes/samples/scripts/create-weblogic-domain/model-in-image/utils/wl-pod-wait.sh

```diff
@@ -118,71 +120,61 @@ function tempfile() {
-    local curvar=colwidth$colcur
-    local width=$(( ${!curvar} ))
-
-    printfexp="${printfexp}%-$((width))s  "
-
-    local pos=0
-    while [ $pos -lt $width ]; do
-      separator="$separator-"
-      pos=$((pos + 1))
-    done
-    separator="$separator "
-
-    colcur=$((colcur+1))
-  done
-  printfexp="${printfexp}\n"
-
-  # now print the table
-  local row=1
-  cat $file | while read line; do
-    printf "$printfexp" $line
-    if [ $row -eq 1 ]; then
-      printf "$printfexp" $separator
-    fi
-    row=$((row + 1))
+function sortlist() {
+  # sort a comma or space separated list
+  #   - stdin input, stdout output
+  #   - spaces replaced with commas
+  #   - blank fields ignored
+  #   - output removes any trailing comma
+  #   - examples: ""->"" "c,b"->"b,c" "c,b,"->"b,c" "c b"->"b,c"
+  tr ' ' '\n'    | \
+  tr ',' '\n'    | \
+  sort -V        | \
+  xargs echo -n  | \
+  tr ' ' ','
+}
+function sortCMImages() {
+  # sort "cmimages=;im2,im1;" field assuming comma or space sep list
+  #   - stdin input, stdout output
+  #   - spaces replaced with commas
+  #   - input ignores trailing comma, output removes any trailing comma
+  #   - examples: see sortCMImagesUnitTest()
+  while read line
+  do
+    echo -n "$line" | sed 's/\(^.*cmimages=;\).*/\1/'
+    echo -n "$line" | sed 's/.*cmimages=;\([^;]*\).*/\1/' | sortlist
+    echo "$line"    | sed 's/.*cmimages=;[^;]*\(;.*\)/\1/'
   done
 }
+function _sortCMImagesUnitTest() {
+  local res=$(echo "$1" | sortCMImages)
+  if [ ! "$res" = "$2" ]; then
+    echo "unit test fail"
+    echo " input ='$1'"
+    echo " expect='$2'"
+    echo " actual='$res'"
+    exit 1
+  fi
+}
+function sortCMImagesUnitTest() {
+  _sortCMImagesUnitTest "foo=;bar; cmimages=;c,b; bar=;foo;"   "foo=;bar; cmimages=;b,c; bar=;foo;"
+  _sortCMImagesUnitTest "foo=;bar; cmimages=; c,b,; bar=;foo;" "foo=;bar; cmimages=;b,c; bar=;foo;"
+  _sortCMImagesUnitTest "foo=;bar; cmimages=;; bar=;foo;"      "foo=;bar; cmimages=;; bar=;foo;"
+  _sortCMImagesUnitTest "foo=;bar; cmimages=;a ; bar=;foo;"    "foo=;bar; cmimages=;a; bar=;foo;"
+  _sortCMImagesUnitTest "cmimages=;c b; bar=;foo; foo=;bar;"   "cmimages=;b,c; bar=;foo; foo=;bar;"
+  _sortCMImagesUnitTest "bar=;foo; foo=;bar; cmimages=; c b ;" "bar=;foo; foo=;bar; cmimages=;b,c;"
+  _sortCMImagesUnitTest "cmimages=;;"                          "cmimages=;;"
+  _sortCMImagesUnitTest "cmimages=; ;"                         "cmimages=;;"
+  _sortCMImagesUnitTest "cmimages=;,,;"                        "cmimages=;;"
+  return 0
+}
+sortCMImagesUnitTest
+
 
-#
-# get domain value specified by $1:
-#   for example '.spec.introspectVersion' or '.spec.restartVersion'
-# and place result in the env var named by $2
-# if expected>0 and echo an Error and exit script non-zero
-#
 function getDomainValue() {
+  # get domain value specified by $1 and put in env var named by $2
+  #  if get fails, and global expected is >0, then echo an Error and exit script non-zero
```

Comment:
minor:  remove space to align description?

Reply:
done